bjs2-70827

tanufa25 opened the pull request:
Stream API task is done

Commit:
bjs2-70827
Stream API task is done

MarselYsup commented on src/main/java/school/faang/bjs2_70827/stream_api1/ListOperations.java:
```diff
@@ -0,0 +1,69 @@
+                .filter(num -> num > number)
+                .min(Comparator.naturalOrder())
+                .orElseThrow(() ->
+                        new NoSuchElementException("There is impossible to find min greater than "
```
Используй форматирование через printf, чтобы не создать большое кол-во объектов в StringPool(так как String неизменяемый и каждое складывание строк создает в памяти новую строку), а также такой подход более производительный, его проще поддерживать, особенно в случае сложных строк или когда выходные данные включают несколько типов данных

```java
 new NoSuchElementException("There is impossible to find min greater than %d. List is empty".formatted(number));
 ```

MarselYsup commented on src/main/java/school/faang/bjs2_70827/stream_api1/Main.java:
```diff
@@ -0,0 +1,31 @@
+@Slf4j
+public class Main {
+    public static void main(String[] args) {
+        final List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6);
```
Можно создать последовательность чисел более лаконично:

```java
List<Integer> range = IntStream.rangeClosed(start, end).boxed().toList();
```

Commit:
bjs2-70827
String formatted method is used instead of concatenation of strings
IntStream.rangeClosed is used for creation of list with sequence of numbers
